commons/pac-batch-commons/src/main/java/com/tmobile/pacman/commons/utils/NotificationPermissionUtils.java

```diff
@@ -53,28 +50,41 @@ public static void triggerNotificationForPermissionDenied(List<PermissionVH> per
             LOGGER.error("Error triggering lambda function url, notification request not sent. Error - {}", e.getMessage());
         }
     }
-    private static NotificationBaseRequest getNotificationBaseRequest(List<PermissionVH> permissionVHList, String cloudType) {
+    private static NotificationBaseRequest getNotificationBaseRequest(JSONObject payload, String cloudType) {
         NotificationBaseRequest notificationBaseRequest = new NotificationBaseRequest();
         notificationBaseRequest.setEventCategory(Constants.NotificationTypes.PERMISSION);
         notificationBaseRequest.setSubject(SUBJECT);
         notificationBaseRequest.setEventCategoryName(Constants.NotificationTypes.PERMISSION.getValue());
         notificationBaseRequest.setEventName(String.format(String.format(opsEventName, cloudType)));
         notificationBaseRequest.setEventDescription(String.format(String.format(opsEventName, cloudType)));
         notificationBaseRequest.setEventSourceName(cloudType);
-        JSONObject permissionNotificationRequest = getPermissionNotificationRequest(cloudType, permissionVHList);
-        notificationBaseRequest.setPayload(permissionNotificationRequest);
+        notificationBaseRequest.setPayload(payload);
         return notificationBaseRequest;
     }
-    private static JSONObject getPermissionNotificationRequest(String cloudType, List<PermissionVH> permissionVHList) {
+    private static List<NotificationBaseRequest> getPermissionNotificationRequest(String cloudType, List<PermissionVH> permissionVHList) {
+        int i=0;
+        List<NotificationBaseRequest> notificationDetailsList = new ArrayList<>();
         JSONObject payload = new JSONObject();
-        payload.put("cloudType", cloudType);
-        payload.put("message", "Unable to collect data due to missing permission");
         for (PermissionVH permissionVH : permissionVHList) {
             for (Map.Entry<String, List<String>> assetPermission : permissionVH.getAssetPermissionIssues().entrySet()) {
+                i++;
                 payload.put("permissionIssueFor" + assetPermission.getKey().toUpperCase() + "InAccount" + permissionVH.getAccountNumber(), assetPermission.toString());
+                if(i%numberOfPermissionIssuesPerEmail==0)
```

Comment:
a bit of nitpicking, but I would get rid of division here and just reset `i` value to 0 inside `if`. Just because simple comparison is faster than division + comparison.

Reply:
"%" operator is to group issues by 40

Reply:
I understand the purpose of `%`, it's basically devision.
My suggestion was to replace division with comparison, and so this if with
```
if (i == 40) {
  ...
  i = 0
}
```
But with our load we will never see the difference between comparison and division performance. 
Ignore it, thinking out loud :)